lib/agent.js

```diff
@@ -24,53 +25,99 @@ class Agent extends EventEmitter {
     }
 
     this[kFactory] = (origin) => factory(origin, opts)
-    this[kCache] = new Map()
+    this[kClients] = new Map()
 
     const agent = this
 
-    this[kOnConnect] = function onConnect (client) {
-      agent.emit('connect', client)
+    this[kOnConnect] = function onConnect (origin, clients) {
+      agent.emit('connect', origin, [...clients, agent])
     }
 
-    this[kOnDisconnect] = function onDestroy (client, err) {
+    this[kOnDrain] = function onDrain (origin) {
+      agent.emit('drain', origin)
+    }
+
+    this[kOnDisconnect] = function onDestroy (origin, clients, err) {
       if (this.connected === 0 && this.size === 0) {
         this.off('disconnect', agent[kOnDisconnect])
-        agent[kCache].delete(this.origin)
+        agent[kClients].delete(this.origin)
       }
 
-      agent.emit('disconnect', client, err)
+      agent.emit('disconnect', origin, [...clients, agent], err)
+    }
+  }
+
+  get connected () {
+    let ret = 0
+    for (const { connected } of this[kClients].values()) {
+      ret += connected
     }
+    return ret
   }
 
-  get (origin) {
-    if (typeof origin !== 'string' || origin === '') {
+  get size () {
```

Comment:
Ok, so the stats needed are directly exposed from the Agent, am I getting this right?

Reply:
Not sure I understand the question?

Reply:
Regarding the fact that `get` allowed to gather stats about a `Client`/`Pool` (https://github.com/nodejs/undici/issues/616#issuecomment-806504163), this new strategy exposes the stats directly on the `Agent`.